test/Expression/Differentiator/Differentiate.purs

```diff
@@ -0,0 +1,164 @@
+module Test.Expression.Differentiator.Differentiate
+  ( differentiateTests
+  ) where
+
+import Prelude
+
+import Data.Either (Either(..))
+import Expression.Differentiator (differentiate)
+import Expression.Error (Expect, throw)
+import Expression.Parser (parse)
+import Expression.Simplifier (simplify)
+import Expression.Syntax (Expression)
+import Test.Unit (TestSuite, suite, test)
+import Test.Unit.Assert (equal)
+
+differentiateTests :: TestSuite
+differentiateTests =
+  suite "Expression.Differentiator - differentiate" do
+    test "ASSERT f(x)' = 0 WHEN f(x) = 1" do
+      let
+        -- given
+        rawExpression = "1"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "0"
+      equal expectedResult result
+    test "ASSERT f(x)' = 1 WHEN f(x) = x" do
+      let
+        -- given
+        rawExpression = "x"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "1"
+      equal expectedResult result
+    test "ASSERT f(x)' = 2*x WHEN f(x) = x^2" do
+      let
+        -- given
+        rawExpression = "x^2"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "2*x"
+      equal expectedResult result
+    test "ASSERT f(x)' = 12*x WHEN f(x) = 6*(x^2)" do
+      let
+        -- given
+        rawExpression = "6*(x^2)"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "12*x"
+      equal expectedResult result
+    test "ASSERT f(x)' = (x^(x--1))*((x)+((x*x)*(logx))) WHEN f(x) = x^x" do
+      let
+        -- given
+        rawExpression = "x^x"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "(x^(x--1))*(x+((x*x)*(logx)))"
+      equal expectedResult result
+    test "ASSERT f(x)' = (-1)/(x^2) WHEN f(x) = 1/x" do
+      let
+        -- given
+        rawExpression = "1/x"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "(-1)/(x^2)"
+      equal expectedResult result
+    test "ASSERT f(x)' = 3*(e^(3*x)) WHEN f(x) = e^(3*x)" do
+      let
+        -- given
+        rawExpression = "e^(3*x)"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "3*(e^(3*x))"
+      equal expectedResult result
+    test "ASSERT f(x)' = e^3 WHEN f(x) = e^3" do
+      let
+        -- given
+        rawExpression = "e^3"
+
+        -- when
+        result = fromExpect $ parseAndDifferentiate rawExpression
+
+        -- then
+        expectedResult = "e^3"
```

Comment:
If f(x) = e^3, then f'(x) = 0 as e^3 is just a constant.

Reply:
Resolved in latest commit